lib/core/test/integration/Test/Integration/Framework/DSL.hs

```diff
@@ -434,6 +436,34 @@ expectPathEventuallyExist filepath = do
             False ->
                 threadDelay (oneSecond `div` 2) >> doesPathExistNow
 
+-- | A file is eventually created on the given location
+expectProcStdOutHas :: CreateProcess -> Text -> IO ()
+expectProcStdOutHas procc out = do
+    handle <- async (runProcUntil procc out)
+    winner <- race (threadDelay (60 * oneSecond)) (wait handle)
+    case winner of
+        Left _ -> expectationFailure $
+            "waited more than 60s for proccess to have in stdout = " ++ T.unpack out
+        Right _ ->
+            return ()
+  where
+    oneSecond = 1_000_000
+
+    runProcUntil :: CreateProcess -> Text -> IO ()
+    runProcUntil pr wants = withCreateProcess pr $ \_ (Just stdout) _ h -> do
+           hSetBuffering stdout LineBuffering
+           waitForIt stdout h
+     where
+         waitForIt stdout h = do
+             is <- TIO.hGetLine stdout
+             case (wants `T.isInfixOf` is) of
+                 True -> do
+                     terminateProcess h
+                     return ()
```

Comment:
suggestion: `void $ terminateProcess h`

Reply:
actually just `terminateProcess h` is enough...